src/Finder/RemovableRedirectFinder.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusRedirectPlugin\Finder;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Setono\SyliusRedirectPlugin\Model\RedirectInterface;
+use Setono\SyliusRedirectPlugin\Repository\RedirectRepositoryInterface;
+use Setono\SyliusRedirectPlugin\Resolver\RedirectionPathResolverInterface;
+
+final class RemovableRedirectFinder implements RemovableRedirectFinderInterface
+{
+    /** @var RedirectRepositoryInterface */
+    private $redirectRepository;
+
+    /** @var RedirectionPathResolverInterface */
+    private $redirectionPathResolver;
+
+    public function __construct(RedirectRepositoryInterface $redirectRepository, RedirectionPathResolverInterface $redirectionPathResolver)
+    {
+        $this->redirectRepository = $redirectRepository;
+        $this->redirectionPathResolver = $redirectionPathResolver;
+    }
+
+    public function findNextRedirect(RedirectInterface $redirect): Collection
```

Comment:
Why return `Collection` instead of just `array`

Reply:
Oh, you use the `contains` method

Reply:
I do prefer Collections over array since the manipulation is indeed way easier.